ts/src/heap-profiler.ts

```diff
@@ -105,3 +113,67 @@ export function stop() {
     stopSamplingHeapProfiler();
   }
 }
+
+export type NearHeapLimitCallback = (profile: Profile) => void;
+
+export const CallbackMode = {
+  Async: 1,
+  Interrupt: 2,
+  Both: 3,
+};
+
+/**
+ * Add monitoring for v8 heap, heap profiler must already be started.
+ * When an out of heap memory event occurs:
+ *  - an extension of heap memory of |heapLimitExtensionSize| bytes is
+ *    requested to v8. This extension can occur |maxHeapLimitExtensionCount|
+ *    number of times. If the extension amount is not enough to satisfy
+ *    memory allocation that triggers GC and OOM, process will abort.
+ *  - heap profile is dumped as folded stacks on stderr if
+ *    |dumpHeapProfileOnSdterr| is true
+ *  - heap profile is dumped in temporary file and a new process is spawned
+ *    with |exportCommand| arguments and profile path appended at the end.
+ *  - |callback| is called. Callback can be invoked only if
+ *    heapLimitExtensionSize is enough for the process to continue. Invocation
+ *    will be done by a RequestInterrupt if |callbackMode| is Interrupt or Both,
+ *    this might be unsafe since Isolate should not be reentered
+ *    from RequestInterrupt, but this allows to interrupt synchronous code.
+ *    Otherwise the callback is scheduled to be called asynchronously.
+ * @param heapLimitExtensionSize - amount of bytes heap should be expanded
+ *  with upon OOM
+ * @param maxHeapLimitExtensionCount - maximum number of times heap size
+ *  extension can occur
+ * @param dumpHeapProfileOnSdterr - dump heap profile on stderr upon OOM
+ * @param exportCommand - command to execute upon OOM, filepath of a
+ *  temporary file containing heap profile will be appended
+ * @param callback - callback to call when OOM occurs
+ * @param callbackMode
+ */
+export function monitorOutOfMemory(
+  heapLimitExtensionSize: number,
+  maxHeapLimitExtensionCount: number,
+  dumpHeapProfileOnSdterr: boolean,
+  exportCommand?: Array<String>,
+  callback?: NearHeapLimitCallback,
+  callbackMode?: number
+) {
+  if (!enabled) {
+    throw new Error(
+      'Heap profiler must already be started to call monitorOutOfMemory'
+    );
+  }
+  let new_callback;
```

Comment:
Minor stylistic nit, but JS generally always does camelCase variable names. Feel free to disregard though. 😅

Reply:
Good catch, I tried to be consistent but this one slipped through ;)